test/check-exercises.py

```diff
@@ -47,7 +47,8 @@ def load_config():
         raise SystemExit(1)
 
     try:
-        problems = [entry['slug'] for entry in data['exercises']]
+        problems = [entry['slug'] for entry in data['exercises']
+                    if "deprecated" not in entry and "foregone" not in entry]
```

Comment:
Foregone exercises are actually listed under a separate list in `config.json`, and do not use the `foregone` field. It won't hurt to leave this, but foregone exercises are already not run.

Reply:
Ah, I presumed that they were now listed in the same way as `deprecated` exercises - I've removed the conditional.